enhancements/olm-admission-webhooks.md

```diff
@@ -0,0 +1,134 @@
+---
+title: olm-admission-webhook-support
+authors:
+  - "@awgreene"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-03-12
+last-updated: 2020-03-12
+status: implementable
+---
+
+# OLM Admission Webhook Support
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement covers the necessary steps to add the capability for OLM to support [validating](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook) and [mutating](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook) admission webhooks.
+
+## Motivation
+
+More and more Operators are shipping webhooks to deliver on core parts of their functionality. Today, operators shipped with OLM are required to manually create and manage their webhooks independently of OLM due to lack of support. In this process, each operator re-creates the logic to implement a webhook and the logic to rotate and manage the coresponding certificates. OLM can help automate much of this process and provide a first-class experience to users, further strengthen OLM's value proposition.
+
+### Goals
+
+- Operator authors can include validating and mutating admission webhooks in their bundle as a part of a CSV.
```

Comment:
Will admission webhooks work out-of-the-box under the [CSVless bundle proposal](https://github.com/operator-framework/enhancements/pull/8)?

Reply:
It is not planned for V1 but could be pulled in depending on how quickly core deliverable are delivered

Reply:
I will call this out in `Non-Goals`.

Reply:
I'm not sure why it wouldn't work, since we're synthesizing a CSV at runtime on the CSVless bundle side.

Reply:
We just need to be sure to add the ability to parse webhook configs into CSVs at bundle build.